messaging/src/test/java/org/axonframework/messaging/annotation/AnnotatedHandlerInspectorTest.java

```diff
@@ -148,6 +150,14 @@ void testInterceptors() throws Exception {
         assertThrows(MockException.class, () -> chain.handle(testEventTwo, testTarget, resultHandler));
     }
 
+    @Test
+    void testGetAllInspectedTypes() {
+        Set<Class<?>> expectedInspectedTypes = Sets.newSet(pA.class, A.class, B.class, C.class, D.class);
+
+        inspector.getAllInspectedTypes()
+                 .forEach(inspectedType -> assertTrue(expectedInspectedTypes.contains(inspectedType)));
```

Comment:
Nit: Would be good also verify the count of inspected types, or reverse the assertion to verify that all of the `expectedInspectedTypes` are contained in `getAllInspectedTypes()`.
The current verification allows for `getAllInspectedTypes()` to return only a subset of `expectedInspectedTypes `

Reply:
Couldn't agree more with you here. We'll make it so!